src/wkt_macro.rs

```diff
@@ -0,0 +1,1020 @@
+/// Creates a [`crate::geometry`] from a
+/// [WKT](https://en.wikipedia.org/wiki/Well-known_text_representation_of_geometry) literal.
+///
+/// This is evaluated at compile time, so you don't need to worry about runtime errors from invalid
+/// WKT syntax.
+///
+/// Note that `POINT EMPTY` is not accepted because it is not representable as a `geo_types::Point`.
+///
+/// ```
+/// use geo_types::wkt;
+/// let point = wkt! { POINT(1.0 2.0) };
+/// assert_eq!(point.x(), 1.0);
+/// assert_eq!(point.y(), 2.0);
+///
+/// let geometry_collection = wkt! {
+///     GEOMETRYCOLLECTION(
+///         POINT(1.0 2.0),
+///         LINESTRING EMPTY,
+///         POLYGON((0.0 0.0,1.0 0.0,1.0 1.0,0.0 0.0))
+///     )
+/// };
+/// assert_eq!(geometry_collection.len(), 3);
+/// ```
+#[macro_export]
+macro_rules! wkt {
+    // Hide distracting implementation details from the generated rustdoc.
+    ($($wkt:tt)+) => {
+        {
+            use $crate::{types::*, Wkt};
+            $crate::wkt_internal!($($wkt)+)
+        }
+    };
+}
+
+#[macro_export(local_inner_macros)]
+#[doc(hidden)]
+macro_rules! wkt_internal {
+    (POINT $tt: tt) => {
+        Wkt::<f64>::Point(point!($tt))
```

Comment:
It'd be nice if this could work for ints too. Is there a reason the macro hardcodes f64?

Is it so that you can do:
```
let point = wkt!(POINT(0.0 0.0))
```

Rather than the more verbose:
```
let point: Wkt = wkt!(POINT(0.0 0.0))
```

Reply:
I think this is resolved with #136. So we can omit the `<f64>` here and someone can define `let point: Wkt<T>` if they need to.

Do you have any feedback on the design of this PR?

Reply:
It looks like if we remove the `::<f64>::`, it infers the type from the value, but also fails for `EMPTY`

<img width="317" alt="image" src="https://github.com/user-attachments/assets/d15a06cd-f0c9-49c5-8ffe-0d7a5c8abaed" />

<img width="330" alt="image" src="https://github.com/user-attachments/assets/5c6df4f2-5f63-45f7-af09-7919e8e6f8c6" />

<img width="384" alt="image" src="https://github.com/user-attachments/assets/cafb1c29-c514-4f42-98b2-4a06492b16ca" />

Reply:
Design wise it seems reasonable. It'd be nice if we could pattern out the various Z/M/ZM variants, but also macros are kind of hard. 

I think this will be a nice feature!

Reply:
> It'd be nice if we could pattern out the various Z/M/ZM variants, but also macros are kind of hard.

Agreed on both counts 🙂.

Despite the verbosity I'll go ahead with fleshing out the rest of the data types.